solara/website/pages/documentation/components/input/file_drop.py

```diff
@@ -10,25 +11,32 @@
 
 @solara.component
 def Page():
-    content, set_content = solara.use_state(b"")
-    filename, set_filename = solara.use_state("")
-    size, set_size = solara.use_state(0)
+    content, set_content = solara.use_state(cast(List[bytes], []))
+    filename, set_filename = solara.use_state(cast(List[str], []))
+    size, set_size = solara.use_state(cast(List[int], []))
+    multiple_upload, set_multiple_upload = solara.use_state(False)
 
-    def on_file(file: FileInfo):
-        set_filename(file["name"])
-        set_size(file["size"])
-        f = file["file_obj"]
-        set_content(f.read(100))
+    def process_multiple_files(files: List[FileInfo]):
+        set_filename([f["name"] for f in files])
+        set_size([f["size"] for f in files])
+        set_content([f["file_obj"].read(100) for f in files])
+
+    def process_single_file(file: FileInfo):
+        process_multiple_files([file])
```

Comment:
Then the two functions that are currently used in the documentation could be combined to one that takes a Union of `FileInfo` and `List[FileInfo]` as an argument, like so:

```suggestion
def process_files(files: Union[FileInfo, List[FileInfo]]):
    if not isinstance(files, list):
        files = [files]
    set_filename([f["name"] for f in files])
    set_size([f["size"] for f in files])
    set_content([f["file_obj"].read(100) for f in files])
```